library/core/src/num/uint_macros.rs

```diff
@@ -984,10 +1009,17 @@ macro_rules! uint_impl {
         #[inline(always)]
         #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
         pub const unsafe fn unchecked_shr(self, rhs: u32) -> Self {
-            // SAFETY: the caller must uphold the safety contract for
-            // `unchecked_shr`.
+            // SAFETY: this is guaranteed to be safe by the caller.
             // Any legal shift amount is losslessly representable in the self type.
-            unsafe { intrinsics::unchecked_shr(self, conv_rhs_for_unchecked_shift!($SelfT, rhs)) }
+            unsafe {
+                let lhs = self;
+                intrinsics::assert_unsafe_precondition!(
+                    concat!(stringify!($SelfT), "::unchecked_shl cannot overflow"),
```

Comment:
```suggestion
                    concat!(stringify!($SelfT), "::unchecked_shr cannot overflow"),
```

Reply:
wait was *this* the reason? I could have *sworn* I double triple checked this